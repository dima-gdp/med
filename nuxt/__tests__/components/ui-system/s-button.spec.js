import { shallowMount } from '@vue/test-utils'
import SButton from '~/components/ui-system/s-button'

describe('ui-system/s-button', () => {
  it('ui-button отображается', () => {
    const wrapper = shallowMount(SButton)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('логировать ошибку, если указать iconName, но не указать icon отличное от "no"', () => {
    const fakeConsoleError = jest.spyOn(console, 'error')

    shallowMount(SButton, {
      propsData: {
        icon: 'no',
        iconName: 'test-name',
      },
    })

    expect(fakeConsoleError).toHaveBeenCalled()
  })
})
