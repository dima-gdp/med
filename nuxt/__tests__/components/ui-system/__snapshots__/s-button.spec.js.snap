// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-system/s-button ui-button отображается 1`] = `<button class="t-b-l s-btn s-btn--c-green s-btn--t-solid"><span class="s-btn__content d-f f-c"></span></button>`;
